feg/gateway/services/envoy_controller/control_plane/control_plane.go

```diff
@@ -121,8 +165,33 @@ func (cb *callbacks) OnFetchRequest(ctx context.Context, req *v2.DiscoveryReques
 	return nil
 }
 
-func (cb *callbacks) OnFetchResponse(*v2.DiscoveryRequest, *v2.DiscoveryResponse) {
-	glog.Infof("OnFetchResponse...")
+func (cb *callbacks) OnFetchResponse(*discovery.DiscoveryRequest, *discovery.DiscoveryResponse) {
+	glog.V(2).Infof("OnFetchResponse...")
+}
+
+func (cb *callbacks) OnDeltaStreamOpen(_ context.Context, id int64, typ string) error {
+	glog.V(2).Infof("delta stream %d open for %s\n", id, typ)
+	return nil
+}
+func (cb *callbacks) OnDeltaStreamClosed(id int64) {
+	glog.V(2).Infof("delta stream %d closed\n", id)
+}
+func (cb *callbacks) OnStreamDeltaResponse(int64, *discovery.DeltaDiscoveryRequest, *discovery.DeltaDiscoveryResponse) {
+	glog.V(2).Infof("OnStreamDeltaResponse...")
+	cb.mu.Lock()
+	defer cb.mu.Unlock()
+	cb.deltaResponses++
+}
+func (cb *callbacks) OnStreamDeltaRequest(int64, *discovery.DeltaDiscoveryRequest) error {
+	glog.V(2).Infof("OnStreamDeltaRequest...")
+	cb.mu.Lock()
+	defer cb.mu.Unlock()
+	cb.deltaRequests++
+	if cb.signal != nil {
+		close(cb.signal)
+		cb.signal = nil
+	}
+	return nil
```

Comment:
_Question:_ Could you say why this is necessary. Not sure, I understand what happens here. :thinking:

Reply:
This, and the other `On...` methods are not called in the code. They are just included to make the `callbacks` type satisfy an interface (`xds.Callbacks`).

Our contribution is to add the four functions that have Delta in the name to the set of `On...` functions. The ones that already existed seem to come from `pkg/test/v3/callbacks.go` (or a predecessor), with some modifications added. We followed that schema to be consistent.